test/spmd/tensor/test_matrix_ops.py

```diff
@@ -127,6 +131,150 @@ def test_t(self):
         self.assertEqual(tranposed_mat2.size(), torch.Size([12, 8]))
         self.assertEqual(tranposed_mat2.placements, shard_spec)
 
+    # PyTorch on cpu seems having issue on baddbmm:
+    # https://github.com/pytorch/pytorch/issues/80588
+    # TODO: Need to investigate why test failed in CPU for baddbmm.
+    @with_comms
+    @skip_if_lt_x_gpu(TEST_GPU_NUM)
+    def test_sharded_baddbmm(self):
+        # If beta is 0, input tensor will be ignored
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        torch.manual_seed(self.rank)
+        tensor = torch.rand(3, 5, 6, device=self.device_type)
+        batch_1 = torch.rand(3, 5, 8, device=self.device_type)
+        batch_2 = torch.rand(3, 8, 6, device=self.device_type)
+        local_result = torch.baddbmm(
+            tensor, batch_1, batch_2, beta=0.0, alpha=0.5
+        )
+        shard0_spec = [Shard(0)]
+        shard1_spec = [Shard(1)]
+        shard2_spec = [Shard(2)]
+
+        def test_comb(
+            tensor_placements: Sequence[Placement],
+            batch_1_placements: Sequence[Placement],
+            batch_2_placements: Sequence[Placement],
+            beta: int,
+            alpha: int,
+        ) -> None:
+            tensor_dt = DTensor.from_local(tensor, device_mesh, tensor_placements)
+            batch_1_dt = DTensor.from_local(batch_1, device_mesh, batch_1_placements)
+            batch_2_dt = DTensor.from_local(batch_2, device_mesh, batch_2_placements)
+            new_dt = torch.baddbmm(
+                tensor_dt, batch_1_dt, batch_2_dt, beta=0.0, alpha=0.5
+            )
+            self.assertEqual(new_dt.to_local(), local_result)
+
+        test_comb(
+            shard0_spec, shard0_spec, shard0_spec,
+            beta=0.0, alpha=0.5
+        )
+
+        test_comb(
+            shard0_spec, shard0_spec, shard0_spec,
+            beta=0.0, alpha=0.5
+        )
+
+        with self.assertRaises(Exception):
+            test_comb(
+                shard1_spec, shard1_spec, shard1_spec,
+                beta=0.8, alpha=0.5
+            )
+
+        with self.assertRaises(Exception):
```

Comment:
can you also make the `test_sharded_baddbmm` follow the same logic in `test_sharded_bmm`?